controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -41,7 +41,7 @@ const InitDiscoveryTopicData = async (req, res) => {
                               FROM posts D 
                               INNER JOIN post_topics E 
                               ON D.post_id = E.post_id 
-                              WHERE E.topic_id = A.topic_id 
+                              WHERE E.topic_id = C.topic_id 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from `E.topic_id = A.topic_id` to `E.topic_id = C.topic_id` alters the join condition of the SQL query. This could potentially lead to different results being returned, depending on the data in your tables. Please ensure that this modification aligns with your intended logic and that `C.topic_id` is available in the scope of this subquery.
```diff
- WHERE E.topic_id = A.topic_id 
+ WHERE E.topic_id = C.topic_id 
```
If `C.topic_id` is not defined in the scope of this subquery, it will result in an error. If it's defined but doesn't have the expected values, it might lead to incorrect results.


<!-- This is an auto-generated comment by OSS CodeRabbit -->